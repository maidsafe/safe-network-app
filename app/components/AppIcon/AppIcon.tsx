import React from 'react';
import { Avatar } from '@material-ui/core';
import FolderIcon from '@material-ui/icons/Folder';

interface Props {
    url?: string;
    fontSize?: 'large' | 'small' | 'default';
    className?: string;
}

export class AppIcon extends React.PureComponent<Props> {
    render() {
        const { url = '', fontSize = 'default', className = '' } = this.props;

        if ( url ) {
            return <Avatar className={className} src={url} />;
        }

        return (
            <Avatar className={className}>
                <FolderIcon fontSize={fontSize} />
            </Avatar>
        );
    }
}
