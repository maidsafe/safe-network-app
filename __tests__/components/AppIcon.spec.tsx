import React from 'react';
import FolderIcon from '@material-ui/icons/Folder';
import { createShallow } from '@material-ui/core/test-utils';
import { AppIcon } from '$Components/AppIcon';

const shallow = createShallow();

const shallowSetup = ( propOverrides? ) => {
    const props = Object.assign(
        {
            url: '',
            fontSize: 'default',
            className: ''
        },
        propOverrides
    );

    const wrapper = shallow( <AppIcon {...props} /> );

    return {
        props,
        wrapper
    };
};

describe( 'App Icon', () => {
    it( 'render', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'by default load folder icon', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper.find( FolderIcon ).exists() ).toBeTruthy();
    } );

    it( 'set url to load image', () => {
        const { wrapper } = shallowSetup( { url: '/some/url/for/img.png' } );
        expect( wrapper.find( FolderIcon ).exists() ).toBeFalsy();
    } );
} );
