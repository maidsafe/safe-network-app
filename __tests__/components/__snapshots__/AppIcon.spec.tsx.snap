// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Icon render 1`] = `ShallowWrapper {}`;
